### PR TITLE
measure image quality

### PR DESCRIPTION
**Pull request by jhung0**

### CODE REVIEW

**Comment by karhohs** [comment]
Please note that TensorFlow is only available in a python 3 flavor for Windows. This has prevented me from testing this plugin on a Windows machine.

**Comment by jhung0** [comment]
Check permissions... some people had  to use sudo to pip install the package and then there was a permission denied error when downloading the weights

**Comment by cells2numbers** [comment]
missing dependency microscopeimagequality, I had to use 

pip install --user microscopeimagequality

to install it.

**Comment by jhung0** [comment]
@0x00b1 @mcquin can you review?

**Comment by AnneCarpenter** [comment]
I'd like to point to the plugin for the blog post Monday, @0x00b1 will it be possible to accept by then? Would be good for it to have a stable URL.

BTW, is there any value in having modules that involve deep learning have a suffix like "-DL" to make them easier to find? Just an idea.

**Comment by AnneCarpenter** [comment]
Kyle points out that FYI TensorFlow is only available in Python3 for Windows, so we should note the incompatibility in the documentation for this module.

**Comment by jhung0** [comment]
@mcquin Could you review?

File: measureimagefocus_requirements.txt [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+scipy
+scikit-image
+numpy
+nose
+matplotlib
+click
+Pillow
+dask
+tensorflow
+microscopeimagequality
```
Comment (mcquin) [inline measureimagefocus_requirements.txt]: `microscopeimagequality` and `matplotlib`* are the only non-CellProfiler requirements that need to be listed in this requirements file. Dependencies of these libraries should automatically be installed and don't need to be explicitly specified.

*matplotlib is a dependency of CellProfiler, and could also be omitted (assuming CellProfiler never drops matplotlib as a dependency)

**Comment by jhung0** [comment]
@mcquin could you review again?

File: measureimagefocus.py [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+    def get_measurement_images(self, pipeline, object_name, category, measurement):
+        if measurement in self.get_measurements(pipeline, object_name, category):
+            return [self.image_name.value]
+
+        return []
+
+    def run(self, workspace):
+        default_weights_index_file = microscopeimagequality.miq.DEFAULT_MODEL_PATH + '.index'
+        if not os.path.exists(default_weights_index_file):
+            print('weights index file not found at {}'.format(default_weights_index_file))
```
Comment (mcquin) [inline measureimagefocus.py]: The standard Python `logging` library should be used in favor of `print` statements.

https://docs.python.org/2/howto/logging.html#a-simple-example

File: measureimagefocus_README.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+Install CellProfiler from source
+Using a terminal (found in Applications), install CellProfiler from source by running given commands in command line (instructions)
+Get CellProfiler Plugin. In the terminal, run these commands:
+```
+MYDIR=/tmp/; cd MYDIR
+git clone https://github.com/CellProfiler/CellProfiler-plugins.git  
+cd CellProfiler-plugins/
+git fetch origin pull/11/head:image-quality
+git checkout image-quality
```
Comment (mcquin) [inline measureimagefocus_README.md]: This line and the line above it won't be necessary when this plugin is merged into master.

File: measureimagefocus_README.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+pip install -r measureimagefocus_requirements.txt
+```
+
+To run CellProfiler, in the terminal run the following; note you may get import errors, but these can be safely ignored:
+```
+cellprofiler --plugins-directory
+```
+
+In CellProfiler,
+Click "Images" and drag in images of interest [1]
```
Comment (mcquin) [inline measureimagefocus_README.md]: Maybe these lines are better displayed as a list of steps? Currently, this reads as a single paragraph (markdown ignores single newlines).

File: measureimagefocus_README.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+cd CellProfiler-plugins/
+git fetch origin pull/11/head:image-quality
+git checkout image-quality
+Install dependencies. In the terminal, run
+pip install -r measureimagefocus_requirements.txt
+```
+
+To run CellProfiler, in the terminal run the following; note you may get import errors, but these can be safely ignored:
+```
+cellprofiler --plugins-directory
```
Comment (mcquin) [inline measureimagefocus_README.md]: You'll need to specify the directory. E.g., `cellprofiler --plugins-directory .` or `cellprofiler --plugins-directory MYDIR/CellProfiler-plugins`

File: measureimagefocus.py [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+        image = cellprofiler.measurement.IMAGE
+
+        features = [
+            self.get_feature_name(F_SCORE),
+            self.get_feature_name(F_CERTAINTY)
+        ]
+
+        column_type = cellprofiler.measurement.COLTYPE_INTEGER
+
+        return [(image, feature, column_type) for feature in features]
```
Comment (mcquin) [inline measureimagefocus.py]: The output I see in the GUI shows the certainty as a Float, rather than an Integer. Is this correct?

Comment (jhung0) [reply measureimagefocus.py]: Yes, should be a float

**Comment by jhung0** [comment]
@mcquin Thanks! Made those changes

**Pull request merged**